File: src/pago.py
"""Módulo que define la clase abstracta Pago."""

from abc import ABC, abstractmethod
from datetime import date


class Pago(ABC):
    """Clase abstracta que representa un pago."""

    def __init__(self, id_pago: int, monto: float, fecha: date):
        self.__id_pago = id_pago
        self.__monto = monto
        self.__fecha = fecha

    @abstractmethod
    def procesar(self) -> bool:
        pass

    @abstractmethod
    def obtener_estado(self) -> str:
        pass

    # --- Getter ---
    def get_id_pago(self) -> int:
        return self.__id_pago

    def get_monto(self) -> float:
        return self.__monto

    def get_fecha(self) -> date:
        return self.__fecha

File: Televentas Java/Clases/Pago.java
package Clases;

import java.time.LocalDate;

public abstract class Pago {

    private int idPago;
    private double monto;
    private LocalDate fecha;

    public Pago(int idPago, double monto, LocalDate fecha) {
        this.idPago = idPago;
        this.monto = monto;
        this.fecha = fecha;
    }

    public abstract boolean procesar();
    public abstract String obtenerEstado();

    // --- Getters ---
    public int getIdPago() { return idPago; }
    public double getMonto() { return monto; }
    public LocalDate getFecha() { return fecha; }
}
